packages/http-client-java/generator/http-client-generator-core/src/main/java/com/microsoft/typespec/http/client/generator/core/mapper/ResponseTypeFactory.java

```diff
@@ -57,26 +57,28 @@ static IType createAsyncResponse(Operation operation, IType bodyType, boolean is
 
             final boolean typedHeadersDisallowed = ignoreTypedHeaders || settings.isDisableTypedHeadersMethods();
             if (typedHeadersDisallowed) {
-                return isByteStream(bodyType) ? mono(ClassType.STREAM_RESPONSE) : mono(GenericType.Response(bodyType));
+                return isByteStream(bodyType)
+                    ? mono(GenericType.Response(ClassType.BINARY_DATA))
+                    : mono(GenericType.Response(bodyType));
             }
```

Comment:
question: if typedHeaderDisallowed=false, what the output class be?

Reply:
It'll be `ResponseBase<HeaderSchema, BinaryData>(since now bodyType cannot be FluxBB):
https://github.com/microsoft/typespec/blob/ba807d9b6095cde876be89d870e79f524d917f50/packages/http-client-java/generator/http-client-generator-core/src/main/java/com/microsoft/typespec/http/client/generator/core/mapper/ResponseTypeFactory.java#L63-L68